haskell/haddock.bzl

```diff
@@ -257,16 +257,26 @@ def _haskell_doc_rule_impl(ctx):
         html_dict_copied[package_id] = output_dir
 
         ctx.actions.run_shell(
-            inputs = [html_dir],
+            inputs = [html_dir] + ctx.files.extra_files,
             outputs = [output_dir],
             command = """
       mkdir -p "{doc_dir}"
       # Copy Haddocks of a dependency.
       cp -R -L "{html_dir}/." "{target_dir}"
+      {extra}
```

Comment:
Putting this here means that the extra files will end-up in the documentation of every package included in this `haskell_doc` target (which can mean a lot if you have enough cabal dependencies).

I guess a way to do this in a more granular way would be to add a `haddock_extra_files` argument to `haskell_{library,binary}` and propagate these extra files in the `haskell_doc_aspect`